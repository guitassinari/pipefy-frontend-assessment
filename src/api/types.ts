export type Pipe = {
  name: string
  color: string
  cards_count: number
}