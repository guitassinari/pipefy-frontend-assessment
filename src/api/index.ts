import Queries from './queries'
import * as Types from './types'

export {
  Queries,
  Types
}