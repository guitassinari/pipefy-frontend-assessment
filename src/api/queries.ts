import { gql } from '@apollo/client'

const GET_ORGANIZATION = gql`
  query GetOrganization($id: ID!) {
    organization(id: $id) {
      id
      name
      pipes {
        id
        name
        color
        cards_count
      }
    }
  }
`

const GET_PIPE_CARDS = gql`
  query GetPipeCards($pipeId: ID!) {
    cards(pipe_id: $pipeId) {
      edges {
        node {
          title
        }
      }
    }
  }
`

export default {
  GET_ORGANIZATION,
  GET_PIPE_CARDS
}