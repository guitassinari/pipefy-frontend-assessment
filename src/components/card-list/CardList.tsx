import React, { FC } from 'react';
import {  Types } from 'api'
import CardTile from 'components/card-tile';
import { CardListContainer } from './CardList.styles'

interface CardListProps {
  cards: Types.Card[]
}

const CardList: FC<CardListProps> = ({ cards }) => (
  <CardListContainer role="grid">
    {cards.map((card: Types.Card, index: number) => (
      <CardTile role="gridcell" key={index} card={card} />
    ))}
  </CardListContainer>
)


export default CardList
