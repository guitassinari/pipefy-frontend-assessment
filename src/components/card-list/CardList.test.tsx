import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CardList from '.'
import { Types } from 'api';

describe('PipeList', () => {
  describe('When receives a valid list of pipes', () => {
    const cards: Types.Card[] = [{
      title: 'a card'
    }, {
      title: 'another card'
    },  {
      title: 'YET another card'
    }]

    it('matches snapshot', () => {
      const component = render(<CardList cards={cards}/>)

      expect(component.asFragment()).toMatchSnapshot()
    })

    it('shows a list of the organization pipes', async () => {
      const { getByText } = render(<CardList cards={cards} />)

      cards.forEach(card => {
        expect(getByText(card.title)).toBeInTheDocument()
      })
    })
  })
})
