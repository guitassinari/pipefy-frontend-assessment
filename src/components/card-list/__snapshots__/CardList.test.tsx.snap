// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipeList When receives a valid list of pipes matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="sc-gsDKAQ fqXist"
    role="grid"
  >
    <div
      class="sc-bdvvtL flRYvv"
      role="gridcell"
    >
      <div>
        <div>
          a card
        </div>
      </div>
    </div>
    <div
      class="sc-bdvvtL flRYvv"
      role="gridcell"
    >
      <div>
        <div>
          another card
        </div>
      </div>
    </div>
    <div
      class="sc-bdvvtL flRYvv"
      role="gridcell"
    >
      <div>
        <div>
          YET another card
        </div>
      </div>
    </div>
  </ul>
</DocumentFragment>
`;
